### DISCUSSION TITLE
Major refactor for improving stability

### DISCUSSION THREAD

**Discussion by doodlewind** [toeverything/AFFiNE#380] [General]
If you've used AFFiNE or joined our social media communities you are sure to know that our product lives up to being an *Alpha* version, having its fair share of bugs. So in order to substantially improve stability and enable users to edit documents smoothly, the AFFINE team are working on a major refactoring of both the paper mode (codename Virgo) and edgeless mode (codename Phasor).

The new technical prototype is still being tuned in our internal branch, but we promise to merge it into the current public repository within the next few weeks. 

There are some current and open issues that will be solved by this refactor so these issues may not be processed individually, but will rather be addressed in the next major release.

Please bear with us as we continue to address your issues and implement both bug fixes and new feature requests 🙏

It's also worth noting that some less commonly used features in the current live demo may be missing in the initial release of the refactored version. We will work hard on building up compatibility in the new editors and implementing these features back in as soon as possible. 

Thanks everyone for trying AFFiNE, we are eager to create a stable and reliable product that can meet your daily requirements ASAP! 

If there are any questions or feedback, please feel free to post here and we'll do our best to answer.

**Comment by Jengro777** [comment]
It's been a month since the core refactoring and I'd like to know roughly when I'll be able to use it

**Reply by ShortCipher5** [reply]
This next week.

**Reply by Jengro777** [reply]
> This next week.
Has affine stoped updating？

**Reply by HeJiachen-PM** [reply]
> > This next week.
> > Has affine stoped updating？

Nope, we are still on a major refactoring. You may have a glance at the alpha version at pathfinder.affine.pro and find the source code in blocksuite repo. Current affine is being progressively migrated.
